### ISSUE TITLE
As a developer, I want to be able to start Glassfish from Netbeans on Mac OS X Mojave (10.14)

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#5431]
As I noted at https://github.com/IQSS/dataverse/issues/1983#issuecomment-451260818 I got back from winter break and upgraded my Mac to OS X Mojave (10.14.0). Now, unfortunately, I'm unable to start Glassfish 4.1 from Netbeans 8.2. Here's the error I get:

> GlassFish Server start failed.
> Please check server admin user name and password properties.
> Also please check the server log file for other possible causes.

Here's a screenshot:

![screen shot 2019-01-04 at 9 20 36 am](https://user-images.githubusercontent.com/21006/50692577-67dc9880-1002-11e9-940b-9d6b10765234.png)

Here's what appears in `/usr/local/glassfish4/glassfish/domains/domain1/logs/server.log`

`[2019-01-04T09:15:03.983-0500] [glassfish 4.1] [INFO] [NCLS-CORE-00087] [javax.enterprise.system.core] [tid: _ThreadID=17 _ThreadName=RunLevelControllerThread-1546611298103] [timeMillis: 1546611303983] [levelValue: 800] [[
  Grizzly Framework 2.3.15 started in: 0ms - bound to [/0.0.0.0:7676]]]`
` `
`[2019-01-04T09:15:03.985-0500] [glassfish 4.1] [INFO] [NCLS-BOOTSTRAP-00027] [javax.enterprise.bootstrap] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1546611303985] [levelValue: 800] [[
  Registered com.sun.enterprise.glassfish.bootstrap.osgi.EmbeddedOSGiGlassFishImpl@69d3cf7e as OSGi service registration: org.apache.felix.framework.ServiceRegistrationImpl@3069a360.]]`
` `
`[2019-01-04T09:15:13.955-0500] [glassfish 4.1] [INFO] [NCLS-JMX-00005] [javax.enterprise.system.jmx] [tid: _ThreadID=61 _ThreadName=Thread-14] [timeMillis: 1546611313955] [levelValue: 800] [[
  JMXStartupService has started JMXConnector on JMXService URL service:jmx:rmi://10.251.93.47:8686/jndi/rmi://10.251.93.47:8686/jmxrmi]]`

To install Glassfish in /usr/local I followed our documented procedure at http://guides.dataverse.org/en/4.10/developers/dev-environment.html#install-glassfish but please not that I did *not* run the Dataverse installer. This is just plain Netbeans 8.2. Plain Glassfish 4.1.

I tried blowing away `~/Library/Application Support/NetBeans/8.2` but it didn't seem to help.

The only other Dataverse developer I know on Mojave is @mheppler and he is not having the trouble above. Main difference seems to be that he's running Netbeans 8.0.2. Perhaps he has already patched Mojave to a newer version? I'll try that next. I can also try Netbeans 9 or 10. Any suggestions are welcome!

**Comment by mheppler** [comment]
I'm on OS 10.14.2 it that helps. Happy to shoot you some Netbeans settings if you want to compare. I also updated to Java 8 Update 191 after the OS update.

**Comment by pdurbin** [comment]
@mheppler thanks. I just upgraded to 10.14.2 and I still can't start Glassfish. I'm on Oracle Java 8 1.8.0_74-b02 so I'm obviously not very good about keeping up to date with my version of Java. Next month Oracle will no longer give out free security updates for Java 8 (#4259) so I should probably switch to https://adoptopenjdk.net or some other non-Oracle solution.

I might actually try Netbeans 10 next. I don't want to feel stuck on old versions of Netbeans like we are stuck on Glassfish 4.1 (we still link to #2628 from http://guides.dataverse.org/en/4.10/developers/tips.html#add-glassfish-4-1-as-a-server-in-netbeans by the way).

**Comment by pdurbin** [comment]
https://stackoverflow.com/questions/46459152/cant-chown-usr-local-for-homebrew-in-osx-10-13-high-sierra/48892451#48892451 got me thinking that maybe System Integrity Protection might be a factor: https://support.apple.com/en-us/HT204899

I tried installing Glassfish in `/Users/pdurbin/java/glassfish/4.1/` rather than `/usr/local` but it didn't help. I guess I'll try Netbeans 10.

**Comment by pdurbin** [comment]
I see the same problem in Netbeans 10.0 (Build incubator-netbeans-release-380-on-20181217). Screenshots first:

![screen shot 2019-01-04 at 3 15 35 pm](https://user-images.githubusercontent.com/21006/50709168-e3573d80-1034-11e9-8692-048bfe56e668.png)

![screen shot 2019-01-04 at 3 21 54 pm](https://user-images.githubusercontent.com/21006/50709169-e3efd400-1034-11e9-9259-363a3e456fae.png)

![screen shot 2019-01-04 at 3 27 13 pm](https://user-images.githubusercontent.com/21006/50709280-447f1100-1035-11e9-821a-833540217aa8.png)


Logs:

`[2019-01-04T15:07:23.067-0500] [glassfish 4.1] [INFO] [NCLS-CORE-00087] [javax.enterprise.system.core] [tid: _ThreadID=17 _ThreadName=RunLevelControllerThread-1546632442489] [timeMillis: 1546632443067] [levelValue: 800] [[
  Grizzly Framework 2.3.15 started in: 1ms - bound to [/0.0.0.0:4848]]]`
` `
`[2019-01-04T15:07:23.102-0500] [glassfish 4.1] [INFO] [NCLS-CORE-00087] [javax.enterprise.system.core] [tid: _ThreadID=19 _ThreadName=RunLevelControllerThread-1546632442496] [timeMillis: 1546632443102] [levelValue: 800] [[
  Grizzly Framework 2.3.15 started in: 2ms - bound to [/0.0.0.0:3700]]]`
` `
`[2019-01-04T15:07:23.114-0500] [glassfish 4.1] [INFO] [NCLS-CORE-00017] [javax.enterprise.system.core] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1546632443114] [levelValue: 800] [[
  GlassFish Server Open Source Edition  4.1  (13) startup time : Felix (32,693ms), startup services(630ms), total(33,323ms)]]`
` `
`[2019-01-04T15:07:28.368-0500] [glassfish 4.1] [INFO] [NCLS-CORE-00087] [javax.enterprise.system.core] [tid: _ThreadID=19 _ThreadName=RunLevelControllerThread-1546632442496] [timeMillis: 1546632448368] [levelValue: 800] [[
  Grizzly Framework 2.3.15 started in: 1ms - bound to [/0.0.0.0:7676]]]`
` `
`[2019-01-04T15:07:28.370-0500] [glassfish 4.1] [INFO] [NCLS-BOOTSTRAP-00027] [javax.enterprise.bootstrap] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1546632448370] [levelValue: 800] [[
  Registered com.sun.enterprise.glassfish.bootstrap.osgi.EmbeddedOSGiGlassFishImpl@69d3cf7e as OSGi service registration: org.apache.felix.framework.ServiceRegistrationImpl@3069a360.]]`
` `
`[2019-01-04T15:07:38.315-0500] [glassfish 4.1] [INFO] [NCLS-JMX-00005] [javax.enterprise.system.jmx] [tid: _ThreadID=62 _ThreadName=Thread-14] [timeMillis: 1546632458315] [levelValue: 800] [[
  JMXStartupService has started JMXConnector on JMXService URL service:jmx:rmi://10.251.93.47:8686/jndi/rmi://10.251.93.47:8686/jmxrmi]]`

Because Java EE doesn't "just work" in Netbeans 9 and newer I started editing source/developers/dev-environment.rst with some notes based on my reading of https://blogs.apache.org/netbeans/entry/what-s-happened-to-my :

---

Please note that while Java EE support is enabled by default in Netbeans 8.2 and lower, more recent versions of Netbeans requires you to enable the "Java EE Base" plugin and its dependencies. Click "Tools", then "Plugins", then "Settings". Make "NetBeans 8.2 Plugin Portal" active by checking the box. Click "Available Plugins" and then "Check for Newest". Check the box for "Java EE Base" and click "Install". After restarting the IDE, follow the prompt to install the nb-javac library.

---

There were a couple of errors along the way.

I had to click "Skip" when I got this one:

"Unable to connect to the NetBeans Distribution because of Timeout while opening connection to https://netbeans.apache.org/nb/updates/10.0/updates.xml.gz?unique=096b54a12-fdcf-4c7d-a7a2-5bc6f04c206e_437b52fa-c31e-4dfd-9aa8-f44dcd2e8df0

Check your proxy settings or try again later. The server may be unavailable at the moment. You may also want to make sure that your firewall is not blocking network traffic."

Maybe my old version of Java is to blame for the one below? I have no idea.

"Warning - could not install some modules: Nashorn Integration - No module providing the capability com.oracle.js.parser.implementation could be found. 12 further modules could not be installed due to the above problems."

I'm not sure what to try next. Maybe upgrade Java from Oracle Java 8 1.8.0_74-b02?

At standup @jggautier said I could try installing Netbeans and Glassfish on his laptop running Mojave. It might be nice to have another data point. I'm not sure which version of Netbeans I'd try.

**Comment by pdurbin** [comment]
Another data point. @michbarsinai reports that he upgraded to Mojave and isn't having trouble with Netbeans and Glassfish. He's on Netbeans 9 and 10.

**Comment by pdurbin** [comment]
I tried moving ~/.gfclient aside. Didn't help. I'm running out of ideas. I could try upgrading to a newer version of Java 8.

I signed up for the users list at https://netbeans.apache.org/community/mailing-lists.html and I guess I'll email them on Monday if I can't figure this out. Frustrating.

**Comment by pdurbin** [comment]
I just posted my sad story as "trouble with NetBeans/GlassFish integration on Mac" to the Netbeans mailing list at https://lists.apache.org/thread.html/809b170e4bbd2f87a4353646d801b2b9daec0c82edf8c72ee7c2efba@%3Cusers.netbeans.apache.org%3E

I also joined #netbeans on freenode.

**Comment by pdurbin** [comment]
Andreas on the Netbeans mailing list mentioned Payara 5 so I tried Payara 5.184 (the latest) and got the same result. I followed up about this at https://lists.apache.org/thread.html/7520c959777765621e453bb303728b2bb3d9c5559542489b72ae5be1@%3Cusers.netbeans.apache.org%3E

Over at https://javabot.evanchooly.com/logs/%23netbeans/2019-01-07 sfisque said, "my first guess would be java. maybe reinstall java. i know between various versions, previous installs of java can get 'wonky'" (and @mheppler also suggested a Java upgrade above). I was on Oracle 1.8.0_74-b02 but I just upgraded to AdoptOpenJDK 1.8.0_192-b12 and I get the same results (I tried Netbeans 8.2 with Glassfish 4.1). Here's a screenshot:

![screen shot 2019-01-07 at 10 16 30 am](https://user-images.githubusercontent.com/21006/50776031-e2f1b900-1265-11e9-9110-a0ba4a76c607.png)
